packages/api-sync/source/sync.ts

```diff
@@ -216,4 +201,84 @@ export class Sync implements Contracts.ApiSync.ISync {
 
 		await this.transactionTypeRepositoryFactory().upsert(types, ["type", "typeGroup", "version"]);
 	}
+
+	async #queueDeferredSync(deferredSync: DeferredSync): Promise<void> {
+		// Limit backlog of pending syncs; normal case the backlog is never expected to be full to begin with.
+		const maxBacklog = 1;
+
+		while (this.#queue.size() > maxBacklog) {
```

Comment:
Queue extends EventEmitter. You can listen to "drain" event when. 

Example: 
this.#queue.once("drain", resolve)